python/Ganga/GPIDev/Lib/Job/Job.py

```diff
@@ -2091,20 +2089,19 @@ def __setattr__(self, attr, value):
 
             # reduce duplicate values here, leave only duplicates for LCG,
             # where we can have replicas
-            uniqueValuesDict = []
-            uniqueValues = []
+            uniqueValuesDict = oDict()
```

Comment:
It sounds like the root problem is that the `__eq__` method on `IGangaFile`s is not intelligent enough for two separate objects resolving to the same canonical path. If it were then a simple `if f not in file_list: file_list.append(f)` would suffice.

Since the original code is a bit confusing (and I agree about the explicit `'LCGSEFile'` check) we can stick with an ordered dictionary.